__tests__/test-stash.js

```diff
@@ -157,6 +193,41 @@ describe('stash push', () => {
     const { fs, dir, gitdir } = await makeFixtureStash('pushThree')
     await stashChanges(fs, dir, gitdir, true, true) // with unstaged changes
   })
+
+  it('stash with untracked files - with other changes', async () => {
+    const { fs, dir, gitdir } = await makeFixtureStash('pushUntracked')
+
+    await addUserConfig(fs, dir, gitdir)
+    await fs.write(`${dir}/a.txt`, 'staged changes - a')
+    await fs.write(`${dir}/b.js`, 'staged changes - b')
+
+    await add({ fs, dir, gitdir, filepath: ['a.txt', 'b.js'] })
+
+    const cContentBeforeStash = 'untracked file - c'
+    const dContentBeforeStash = 'untracked file - d'
+
+    // Create untracked files
+    await fs.write(`${dir}/c.txt`, 'untracked file - c')
+    await fs.write(`${dir}/d.js`, 'untracked file - d')
+
+    let error = null
+    try {
+      await stash({ fs, dir, gitdir, op: 'push' })
+    } catch (e) {
+      error = e
+    }
+
+    expect(error).toBeNull()
+    const cContentAfterStash = new TextDecoder().decode(
+      await fs.read(`${dir}/c.txt`)
+    )
+    const dContentAfterStash = new TextDecoder().decode(
+      await fs.read(`${dir}/d.js`)
+    )
```

Comment:
You should be able to use toString() to the output of `fs.read` if you use 'utf8' as the second argument, you will get a string.